web-registry/src/pages/CreditClassDetails/hooks/useImpact.ts

```diff
@@ -1,51 +1,107 @@
 import { useQuery } from '@tanstack/react-query';
-import { useSortResultWithIris } from 'utils/sanity/useSortResultWithIris';
 
-import { getEcologicalImpactByIriQuery } from 'lib/queries/react-query/sanity/getEcologicalImpactByIriQuery/getEcologicalImpactByIriQuery';
+import { ProjectImpactCardProps } from 'web-components/lib/components/cards/ProjectImpactCard/ProjectImpactCard';
+
+import { client as sanityClient } from 'lib/clients/sanity';
+import { CreditClassMetadataLD } from 'lib/db/types/json-ld';
+import { getSdgByIriQuery } from 'lib/queries/react-query/sanity/getSdgByIriQuery/getSdgByIriQuery';
 
 import { Maybe } from '../../../generated/graphql';
-import { EcologicalImpact } from '../../../generated/sanity-graphql';
-import { client } from '../../../lib/clients/sanity';
+import {
+  normalizeCoBenefit,
+  normalizePrimaryImpact,
+} from '../CreditClassDetails.utils';
+import useCoBenefits from './useCoBenefits';
+import usePrimaryImpact from './usePrimaryImpact';
 
 interface InputProps {
-  coBenefitsIRIs: Maybe<string | string[]> | undefined;
-  primaryImpactIRI?: string;
+  offChainCoBenefitsIRIs: Maybe<string | string[]> | undefined;
+  offChainPrimaryImpactIRI?: string;
+  creditClassMetadata?: CreditClassMetadataLD;
 }
 
 export default function useImpact({
-  coBenefitsIRIs,
-  primaryImpactIRI,
-}: InputProps): EcologicalImpact[] {
-  const { data: primaryImpactData } = useQuery(
-    getEcologicalImpactByIriQuery({
-      iris: primaryImpactIRI,
-      sanityClient: client,
-      enabled: !!primaryImpactIRI,
-    }),
-  );
+  offChainCoBenefitsIRIs,
+  offChainPrimaryImpactIRI,
+  creditClassMetadata,
+}: InputProps) {
+  const primaryImpact = creditClassMetadata?.['regen:primaryImpact'];
+  const coBenefits = creditClassMetadata?.['regen:coBenefits'];
 
-  const { data: coBenefitData } = useQuery(
-    getEcologicalImpactByIriQuery({
-      iris: coBenefitsIRIs,
-      sanityClient: client,
-      enabled: !!coBenefitsIRIs,
+  const primaryImpactSdgIris =
+    primaryImpact?.['regen:SDGs']?.map(sdg => sdg['@id']) || [];
+  const sdgIris: string[] = [...primaryImpactSdgIris];
+  let coBenefitsSdgIris: string[][] = [];
+  for (const coBenefit of coBenefits || []) {
```

Comment:
nit; would it be better to move the `|| []` to where `const coBenefits` is declared?